sources/Inputs/src/NetworkManager.cpp

```diff
@@ -98,10 +98,11 @@ NetworkManager::buildTree() {
       // node should exist at this point
       assert(nodes_.count(nodeid));
 #endif
-      if ((*it_g)->isVoltageRegulationOn()) {
+      auto targetP = (*it_g)->getTargetP();
+      if ((*it_g)->isVoltageRegulationOn() && targetP >= (*it_g)->getPMin() && targetP <= (*it_g)->getPMax()) {
         // We don't use dynamic models for generators with voltage regulation disabled
```

Comment:
We don't use dynamic models for generators with voltage regulation disabled and an active power reference outside the generator's PQ diagram